File: backend/questions/template_question/javascript/grader.js
const fs = require("fs");
const solution  = require("./user_solution");

fs.readFile("../input", (err, input) => {
    if (err) throw err;

    fs.readFile("../output", (err, output) => {
        if (err) throw err;

        const inputs = input.toString().split('\n');
        const outputs = output.toString().split('\n');

        var total = outputs.length;
        var correct = 0;

        for (var i = 0; i < total; i++) {
            const result = solution.solve(inputs[i * 2].split(" ").map(Number), parseInt(inputs[i * 2 + 1]));
            if (JSON.stringify(result) === JSON.stringify(outputs[i].split(" ").map(Number))) correct++;
        }

        console.log("%d/%d test cases passed!", correct, total);
    })
});

File: backend/questions/template_question/java/Grader.java
import java.io.*;
import java.util.*;

public class Grader {
  public static void main(String[] args) {
    try {
      File input = new File("../input");
      File output = new File("../output");
      Scanner inputReader = new Scanner(input);
      Scanner outputReader = new Scanner(output);
      int total = 0, correct = 0;
      Solution solution = new Solution();
      while (inputReader.hasNextLine()) {
        total++;
        int[] nums = Arrays.stream(inputReader.nextLine().split(" ", 0)).mapToInt(Integer::parseInt).toArray();
        int target = Integer.parseInt(inputReader.nextLine());
        int[] ans = Arrays.stream(outputReader.nextLine().split(" ", 0)).mapToInt(Integer::parseInt).toArray();
        int[] result = solution.solve(nums, target);
        if (Arrays.equals(result, ans)) correct++;
      }
      inputReader.close();
      outputReader.close();
      System.out.printf("%d/%d test cases passed!\n", correct, total);
    } catch (Exception e) {
      e.printStackTrace();
    }
  }
}